### ISSUE TITLE
Host migration

### ISSUE THREAD

**Issue by jimmydorry** [GetDotaStats/site#232]
We're moving to the land down under!

- [x] Do domain name audit
- [x] Setup VM for DB
- [x] Setup VM for webserver (PHP + apache2 + php-curl + virtual hosts in apache2 for existing domains) OR nginx https://github.com/GetDotaStats/site/issues/192
- [x] Make new sub-domain to test with (12hour propagation initially)
- [x] Test major site functionality and performance. (Site is currently built with assumption of localhost DB. If there is a notable performance degrade from sockets to virtual server, then just build them together.)
- [x] Migrate old DB
- [x] Test site performance with the full DB. Run the big cronjobs to benchmark
- [x] Update dns zone-file to point at new host

**Comment by SinZ163** [comment]
...Why

**Comment by jimmydorry** [comment]
Better and more resources to use.

**Comment by SinZ163** [comment]
at the expense of slower connections for a majority of users

**Comment by jimmydorry** [comment]
Not really a cause for major concern. 

Site performance improvements will have a far larger impact on user experience, than 100-200ms of extra latency.

Having more data storage and more processing power allows us to work with larger data sets, far faster.

**Comment by jimmydorry** [comment]
Letter to modders, sent out on the 22nd of Feb:

--------------------

Salutations modder comrades!

It's been a while since you've last heard from me. We've had a site migration planned for some time now, but it's finally becoming a reality.

**_We will be having some downtime this Wednesday 24th of February._**

The hardware upgrade should give us more breathing room to handle the large site load, not to mention make everything a bit faster. We are currently averaging somewhere around 50 database transactions a second.

**What this means for you modders:**

- There will be an hour or two of down time on Wednesday as we move and wait for the DNS changes to propagate.
- In a week or so, there will be a new version of the library available. The new version will make future upgrades easier to manage (it will point at a subdomain, instead of the main site).
- Before the migration, some stats will not get generated like: Player Values, Games Values, and Win-rates.
- Some matches may not record properly until we have successfully migrated.
- After migration, there will probably be some errors and instability as we tweak and fix things that slipped through our rigorous testing process.
- We have also heard your pleas for some additional features like: most active users and better control of how win rates are broken down. They are still works in progress, but I am making progress.

Feature requests are still welcomed in our issue tracker: http://github.com/GetDotaStats/site/issues

As a bonus, here are some pictures from my last visit to the Data Centre where I installed three-quarters of the RAM and HDDs required in the server. And yes, I fully own this server. 

http://imgur.com/a/8VC8F

Regards,
jimmydorry
GetDotaStats Generál-mayór
